### ISSUE TITLE
Invalid audience for Key Vault secret operations when using MSI

### ISSUE THREAD

**Issue by thomas11** [pulumi/pulumi-azure-native#2432]
```
error: keyvault.BaseClient#DeleteSecret: Failure responding to request: StatusCode=401 -- Original Error: autorest/azure: Service returned an error. Status=401 Code="Unauthorized" Message="AKV10022: Invalid audience. Expected https://vault.azure.net, found: https://management.azure.com/."
```

When using Managed Identity (ARM_USE_MSI) with the Pulumi Automation API. See #2374 for the original report. That issue is about the same error but using az CLI authentication, which was fixed.

**Comment by brucetan-displayr** [comment]
Having the same error, woudl be good if this is fixed

**Comment by thomas11** [comment]
I took a more in-depth look at this and unfortunately, it's due to a shortcoming in a dependency that we use that cannot be upgraded quickly. Related to #1464. We will try to schedule this workstream soon.

**Comment by thecodetinker** [comment]
Hello - we have just tried this with v2.75.0 and unfortunately got a different error. 

`error: keyvault.BaseClient#DeleteSecret: Failure responding to request: StatusCode=401 -- Original Error: autorest/azure: Service returned an error. Status=401 Code="Unauthorized" Message="AKV10022: Invalid audience. Expected cfa8b339-82a2-471a-a3c9-0fc0be7a4093, found: https://management.azure.com/."`

The GUID represents the Key Vault Service Principal - it is registered as the same in all tenants (as per [this doc](https://learn.microsoft.com/en-us/azure/key-vault/secrets/overview-storage-keys#service-principal-application-id)). Not sure if I need to open a new issue for this.

**Comment by thomas11** [comment]
Hi @thecodetinker, sorry for the additional trouble here. This is curious, as we have a CI test for secrets that passes, and I also just verified it locally using `az` user authentication. Maybe it's another MSI-specific issue. I'll look into testing that as soon as I can.

In the meantime, would you open a new issue please? It's helpful for tracking.

Here's my minimal attempted repro.

```yaml
name: azn-2374-delete-secret
runtime: yaml

config:
  tenantId:
    type: string
  objectId:
    type: string

resources:
  resourceGroup:
    type: azure-native:resources:ResourceGroup

  keyVault:
    type: azure-native:keyvault:Vault
    properties:
      resourceGroupName: ${resourceGroup.name}
      properties:
        sku:
          name: standard
          family: A
        tenantId: ${tenantId}
        accessPolicies:
          - tenantId: ${tenantId}
            objectId: ${objectId}
            permissions:
              secrets:
                - get
                - list
                - delete

  secret:
    type: azure-native:keyvault:Secret
    properties:
      resourceGroupName: ${resourceGroup.name}
      vaultName: ${keyVault.name}
      properties:
        value: mySecretValue
        attributes:
          enabled: true

outputs:
  vaultUri: ${keyVault.properties.vaultUri}
```

**Comment by Jojooo0o** [comment]
i am facing the same issues as @thecodetinker running on the v2.70.0 Error occurs only when using MSI as per original report. Is there a new bug that is tracking this now? If so, please share :)

**Comment by thomas11** [comment]
There isn't a new tracking issue yet. Would you mind filing one, with as much detail as possible about your setup?

**Comment by thecodetinker** [comment]
Hi @thomas11 thanks for the reply. I would need to unwind the workarounds to reproduce this for you and I'm not sure I will have time to give you a good bug report before end of year. Suffice to say it's the same issue as before, but with a different error as detailed above. 

For context, we run this via the Automation API in an Azure DevOps pipeline within an **AzureCLI@2** task. We run the pipeline on an agent with an assigned Managed Identity that has access to the relevant resources for deployment. The script sets  `$env:ARM_USE_MSI="true"` and then runs `dotnet Automation.dll` to actually run the pulumi.

Interesting that @Jojooo0o is having the same issue - are you also using the Automation API?

**Comment by Jojooo0o** [comment]
Hey @thomas11 and @thecodetinker,

Hope you had a lovely christmas and new years time :)

I have been trying to reproduce the issue for a new ticket but it seems it was resolved in some version after `v2.70.0`. On `v2.70.0` i get the described error (by @thecodetinker) when running the below script and then rerunning while commenting out the secret (to "delete" it):

```python
import pulumi
import pulumi_azure_native as azure_native
from pulumi.automation import create_or_select_stack, ConfigValue, LocalWorkspaceOptions, ProjectSettings, ProjectBackend

def pulumi_program():
    secret = azure_native.keyvault.Secret("automationexampleSecret",
        resource_group_name='sample-rg',
        vault_name='sample-kv',
        secret_name='automation-secret',
        properties=azure_native.keyvault.SecretPropertiesArgs(
            value='automationsomesecret',
        ),
    )

    pulumi.export("secret_id", secret.id)

def run():
    opts = LocalWorkspaceOptions(
        project_settings=ProjectSettings(name="automation-project", runtime='python', backend=ProjectBackend('azblob://<blob-url>')),
        env_vars={
            'PULUMI_CONFIG_PASSPHRASE': 'some_secret',
            'ARM_USE_MSI': 'true',
            'ARM_SUBSCRIPTION_ID': '<az_sub_id>'
        },
        work_dir = '.'
    )
    stack = create_or_select_stack(
        stack_name='automation-stack',
        project_name='automation-project',
        program=pulumi_program,
        opts=opts)
    stack.set_config("azure-native:location", ConfigValue("westeurope"))
    stack.up(on_output=print)

if __name__ == "__main__":
    run()
```

It seems the error will prevail until you successfully update the stack to use a later version (not sure which, but going from v2.70->2.75 seems to work. For me it looked like the stack needs to be able to update to a newer provider to fix this.

Hope this makes somewhat sense and maybe helps you out as well @thecodetinker.
